src/Nethermind/Nethermind.Blockchain/BlockProcessor.cs

```diff
@@ -186,17 +200,23 @@ private Block ProcessOne(Block suggestedBlock, ProcessingOptions options, IBlock
             }
 
             Block block = PrepareBlockForProcessing(suggestedBlock);
+            _additionalBlockProcessor?.PreProcess(block);
+            
             var receipts = ProcessTransactions(block, options, blockTracer);
```

Comment:
IBlockProcessor: PreProcess, Process, PostProcess and each implementation can skip or apply, then use Composite?

Reply:
Ah I see. Hm... I would prefer have base BlockProcessor and AuraBlockProcessor with some overrides. Maybe that is better?